### PR TITLE
feat: compose functions

### PR DESCRIPTION
**Pull request by karranb**
Basic implementation of pipe and compose.

I'm having a little hard time thinking about the descriptions of the function and I'm also open to implementing more test cases if anyone got some suggestions.

### CODE REVIEW

File: src/common/compose/test.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+describe('compose', () => {
+  it('should validate curried numbers', () => {
+    const add1 = jest.fn((x) => x + 1)
+    const multiply2 = jest.fn((x) => x * 2)
+    const result = compose(add1, multiply2)(1)
+    expect(result).toEqual(3)
+    expect(add1).toBeCalled()
+    expect(multiply2).toBeCalled()
+  })
```
Comment (helderburato) [inline src/common/compose/test.js:11]: I think you can break each `expect` as a different test case, to be more readable and focus each test on just one thing.

File: src/common/pipe/test.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { pipe } from '.'
+
+describe('pipe', () => {
+  it('should validate pipe ', () => {
+    const add1 = jest.fn((x) => x + 1)
+    const multiply2 = jest.fn((x) => x * 2)
+    const result = pipe(add1, multiply2)(1)
+    expect(result).toEqual(4)
+    expect(add1).toBeCalled()
+    expect(multiply2).toBeCalled()
```
Comment (helderburato) [inline src/common/pipe/test.js:10]: Same here.

**Review by helderburato — CHANGES_REQUESTED** [review]
I think the test-cases can be more broken into smaller parts and don't forget to generate the docs. markdown for every implemented function, if you need some help to make this, just ping me 💯 great work!

**Comment by karranb** [comment]
I made a small refactor.
- Instead of breaking the tests into 3 cases, I broke them into 2. (I'm not sure if that's okay)
- I also created an `applyTo` based on `ramda's` one to use on the compose functions.https://ramdajs.com/docs/#applyTo (I'm not sure if that's okay too)

**Review by helderburato — APPROVED** [review]
Amazing! 💯 
Feel free to merge and close the related issues @karranb

**Pull request merged**